xray/commands/audit/npm.go

```diff
@@ -0,0 +1,146 @@
+package scan
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/jfrog/jfrog-cli-core/artifactory/commands/npm"
+	commandsutils "github.com/jfrog/jfrog-cli-core/artifactory/commands/utils"
+	"github.com/jfrog/jfrog-cli-core/utils/config"
+	"github.com/jfrog/jfrog-cli-core/xray/commands"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/jfrog/jfrog-client-go/xray/services"
+)
+
+const (
+	NpmPackageTypeIdentifier = "npm://"
+)
+
+type XrAuditNpmCommand struct {
+	serverDetails    *config.ServerDetails
+	workingDirectory string
+	arguments        []string
+	typeRestriction  npm.TypeRestriction
```

Comment:
```npm.TypeRestriction``` is under the ```artifactory/commands``` path, but we're under ```xrat/commands```.
We should move shared resources to be under a common package, which can be shared between Artifactory and Xray.
I think there are other components that need to be moved, such as ```npm.NewNpmCommandArgs```.